libraries/engage/tracking/actions/cookieConsent.js

```diff
@@ -0,0 +1,72 @@
+import {
+  historyPush,
+  grantAppTrackingTransparencyPermission,
+} from '@shopgate/engage/core/actions';
+import {
+  updateCookieConsent,
+  hideCookieConsentModal,
+} from '../action-creators';
+import { PRIVACY_SETTINGS_PATTERN } from '../constants';
+
+/**
+ * action to be dispatched when the user accepted all cookies in the custom modal
+ * and native modal should be triggered for setting the permission
+ * @returns {Function}
+ */
+export const acceptAllCookies = () => async (dispatch) => {
+  await dispatch(grantAppTrackingTransparencyPermission());
+
+  dispatch(updateCookieConsent({
+    comfortCookiesAccepted: true,
+    statisticsCookiesAccepted: true,
+  }));
+
+  dispatch(hideCookieConsentModal());
```

Comment:
we removed this line in PWA6 recently. it added a new history entry instead of just hiding the modal

https://github.com/shopgate/pwa/pull/1265/files#diff-cc3995fce7e740ff56a08ab9941af0cb4052cf166db187486ead2c8d1802371cL25